src/k2/cmd/tpcc/transactions.h

```diff
@@ -146,33 +156,44 @@ class PaymentT : public TPCCTxn
     }
 
     future<> customerUpdate() {
-        return _txn.read<Customer>(Customer(_c_w_id, _c_d_id, _c_id))
-        .then([this] (auto&& result) {
-            CHECK_READ_STATUS(result);
-
-            Customer customer(_c_w_id, _c_d_id, _c_id);
-            customer.Balance = *(result.value.Balance) - _amount;
-            customer.YTDPayment = *(result.value.YTDPayment) + _amount;
-            customer.PaymentCount = *(result.value.PaymentCount) + 1;
-            k2::String str500(k2::String::initialized_later{}, 500);
-            customer.Info = str500;
-
-            if (*(result.value.Credit) == "BC") {
-                size_t shift_size = sizeof(_c_id) + sizeof(_c_d_id) + sizeof(_d_id) + sizeof(_w_id) + sizeof(_amount);
-                memcpy((char*)customer.Info->c_str() + shift_size, (char*)result.value.Info->c_str(), 500-shift_size);
-                uint32_t offset = 0;
-                memcpy((char*)customer.Info->c_str() + offset, &_c_id, sizeof(_c_id));
-                offset += sizeof(_c_id);
-                memcpy((char*)customer.Info->c_str() + offset, &_c_d_id, sizeof(_c_d_id));
-                offset += sizeof(_c_d_id);
-                memcpy((char*)customer.Info->c_str() + offset, &_d_id, sizeof(_d_id));
-                offset += sizeof(_d_id);
-                memcpy((char*)customer.Info->c_str() + offset, &_w_id, sizeof(_w_id));
-                offset += sizeof(_w_id);
-                memcpy((char*)customer.Info->c_str() + offset, &_amount, sizeof(_amount));
-            }
-
-            return partialUpdateRow<Customer, std::vector<k2::String>>(customer, {"Balance", "YTDPayment", "PaymentCount", "Info"}, _txn).discard_result();
+        uint32_t cid_type = _random.UniformRandom(1, 100);
+        future<int32_t> cidFuture = make_ready_future<int>();
```

Comment:
You need to pass in _c_id here or it will be lost when the future response is saved below.